src/xdb_lib.erl

```diff
@@ -144,19 +144,19 @@ reduce_while(Fun, AccIn, List) when is_function(Fun, 2) ->
   catch
     throw:{halt, AccOut} ->
       AccOut;
-    Kind:Reason ->
-      erlang:raise(Kind, Reason, erlang:get_stacktrace())
+    _:Reason ->
```

Comment:
Let's remove this clause, we don't need it, if it is not `throw :...` bu default an error is raised.

Reply:
@maximvl did you have the chance to address the suggested fix above ^^ ??

Reply:
@cabol sorry for long reply, could you tell which exact part we can remove here? whole catch?

Reply:
I meant you should remove this pattern-match ` _:Reason`

Reply:
@maximvl what about this? For me it should be removed, this pattern match doesn't make sense since by default the original error is raised.